components/PoeModal.tsx

```diff
@@ -0,0 +1,659 @@
+                        methodSchema.Execute(detail.parameters.value)
+                      ).toTransferParams().parameter?.value;
+
+                      if (!value) return undefined;
+
+                      const param = emitMicheline(value as Expr);
+
+                      const parser = new Parser();
+
+                      const type = emitMicheline(
+                        parser.parseJSON(
+                          contract.entrypoints.entrypoints[
+                            detail.parameters.entrypoint
+                          ]
+                        ),
+                        {
+                          indent: "",
+                          newline: "",
+                        }
+                      );
+
+                      return {
+                        type: "contract",
+                        values: {
+                          lambda: generateExecuteContractMichelson(version, {
+                            address: detail.destination,
+                            amount: Number(detail.amount),
+                            entrypoint: detail.parameters.entrypoint,
+                            type,
+                            param,
+                          }),
+                          metadata: null,
+                        },
+                      };
+                    } catch (e) {
+                      const errorMessage = `Failed to create a TzSafe proposal: ${
+                        (e as Error).message
+                      }`;
+                      setTransactionError(errorMessage);
+                      console.log("Contract conversion error:", e);
+
+                      state.p2pClient?.sendError(
+                        message.id,
+                        errorMessage,
+                        BeaconErrorType.TRANSACTION_INVALID_ERROR
+                      );
+                      return undefined;
+                    }
+                  } else
+                    return {
+                      type: "transfer",
+                      values: {
+                        to: detail.destination,
+                        amount: detail.amount,
+                        parameters: {},
+                      },
+                    };
+
+                case TezosOperationType.DELEGATION:
+                  return !!detail.delegate
+                    ? {
+                        type: "lambda",
+                        values: {
+                          lambda: generateDelegateMichelson(version, {
+                            bakerAddress: detail.delegate,
+                          }),
+                          metadata: JSON.stringify({
+                            baker_address: detail.delegate,
+                          }),
+                        },
+                      }
+                    : undefined;
+
+                default:
+                  return undefined;
+              }
+            })
+          )
+        ).filter(v => !!v) as transfer[]
```

Comment:
before `filter` should check if there is `undefined`. If there is, send error back to client.